### DISCUSSION TITLE
ToggleColumn with modal confirmation?

### DISCUSSION THREAD

**Discussion by OvalMedia** [filamentphp/filament#6839] [Help] [answered]
In a table I am using a toggle column to set invoices to a paid status. This is supposed to be a one way status change. Once paid it should be locked in. But without confirmation it is way too easy to accidentally toggle a switch.

Would it be possible to have a modal confirmation like with an action?

(If not: make this a feature request)

Thank you.

**Comment by pboivin** [comment, marked as answer]
Just a thought: if the operation can't be reversed, should it be a toggle? Can you make it an action instead? This way, you would be able to have a confirmation modal.

**Reply by OvalMedia** [reply]
You are right. Once I found out about a toggle column I was too fixated and totally forgot that a simple action is more suitable. Thank you for the slap.

**Reply by OvalMedia** [reply]
Still: an optional confirmation modal for toggle columns might be a useful feature?

**Reply by pboivin** [reply]
It's not a bad idea, considering that input columns modify the record without an explicit "save" action :)

**Reply by gyaanesh** [reply]
confirmation modal for toggle columns should be a useful feature. like I have a toggle for user Is_active and accidentally I clicked its toggle and I am not aware of which user did I clicked as I was not looking at the screen? now how can I undo it ??

**Reply by dangnhdev** [reply]
➡️ `I have a toggle for user Is_active`
A lot of us too I think 😂😂😂

**Comment by superuser-me** [comment]
@OvalMedia 
I am just facing this same issue,
not all people who use admin panels or any panels are smart,
they might accidentally toggle some things on panel,
i really wanted ToggleColumn design with conformation modal.

**Comment by peter-mw** [comment]
Any news ? I have the same problem. I need the user to confirm before toggling.

**Comment by franmesa** [comment]
> In a table I am using a toggle column to set invoices to a paid status. This is supposed to be a one way status change. Once paid it should be locked in. But without confirmation it is way too easy to accidentally toggle a switch.
> 
> Would it be possible to have a modal confirmation like with an action?
> 
> (If not: make this a feature request)
> 
> Thank you.


i solve this with ->beforeStateUpdated and a modal form hope this help.

**Reply by ahmed-alii** [reply]
Doesn't help. Can you maybe give out an example?
